tests/test_allow_zeros.py

```
@@ -0,0 +1,167 @@
+
+
+def test_non_owner_cannot_toggle_allow_zeros(
+    owner_client: TNClient, non_owner_client: TNClient
+):
+    """The owner gate on set_allow_zeros rejects non-owners."""
+    stream_id = generate_stream_id("allow_zeros_owner_gate")
+    _safe_destroy(owner_client, stream_id)
+
+    owner_client.deploy_stream(stream_id)
+    try:
+        with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)